app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package org.coepi.android.ui.symptoms.fever
+
+import androidx.lifecycle.ViewModel
+import org.coepi.android.domain.symptomflow.SymptomFlowManager
+import org.coepi.android.domain.symptomflow.SymptomInputs.Cough
+import org.coepi.android.domain.symptomflow.Temperature
+import org.coepi.android.ui.navigation.NavigationCommand.Back
+import org.coepi.android.ui.navigation.RootNavigation
+
+class FeverHighestTemperatureViewModel(
+    val navigation: RootNavigation,
+    private val symptomFlowManager: SymptomFlowManager
+) : ViewModel() {
+
+    fun onTempChanged(tempStr: String) {
+        if (tempStr.isEmpty()) {
+            symptomFlowManager.setFeverHighestTemperatureTaken(null)
+        } else {
+            val temperature: Float = tempStr.toFloatOrNull() ?: error("Invalid input: $tempStr")
+            symptomFlowManager.setFeverHighestTemperatureTaken(Temperature(temperature))
+        }
+    }
+
+    fun onClickSubmit() {
+        symptomFlowManager.navigateForward()
+    }
+
+    fun onClickUnknown() {
+        symptomFlowManager.navigateForward()
+    }
+
+    fun onClickSkip() {
+        symptomFlowManager.navigateForward()
+    }
+
+    fun onBack() {
+        symptomFlowManager.onBack()
+    }
+
+    fun onBackPressed() {
+        onBack()
+        navigation.navigate(Back)
+    }
+
+    private fun cToF(c: Double) = (9/5.0 * c) + 32
+    private fun fToC(f: Double) = 5/9.0 * (f - 32)
```

Comment:
What about something like this:

```kotlin
sealed class Temperature {
    data class Celsius(val value: Float) : Temperature() {
        override fun toFarenheit(): Farenheit =
            Farenheit(value) // TODO convert
    }
    data class Farenheit(val value: Float) : Temperature() {
        override fun toCelsius(): Celsius =
            Celsius(value) // TODO convert
    }

    open fun toCelsius(): Celsius = when (this) {
        is Celsius -> this
        is Farenheit -> toCelsius()
    }

    open fun toFarenheit(): Farenheit = when (this) {
        is Farenheit -> this
        is Celsius -> toFarenheit()
    }
}
```

Reply:
Wasn't exactly sure how to use this, but put it in place for now, see latest commit. Let me know if that was what you were thinking, or how I should fix it. Thanks!

Reply:
Almost! Now replace the old `Temperature` with this (you've to make it `Parcelable` I think). This way always know the unit it's in.

Reply:
Gotcha, I moved the class into SymptomInputs and replaced the existing temperature class, so now we can pass in Celsius or Fahrenheit objects from the viewmodel.

Reply:
I'd leave it as a standalone class, it's unlikely that we will use it outside the inputs but it's still independent.